### PR TITLE
Added custom fields on the detail page

### PR DESCRIPTION
**Pull request by mesemus**

### CODE REVIEW

File: oarepo_vocabularies/ui/theme/assets/semantic-ui/less/oarepo_vocabularies_ui/definitions/vocabulary_cf.less [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+/*
+  This is the styling of your component. The class should start with .ui.<component-name>
+
+  Note: use variables here, not the exact values. Placing the values to another file (sample.variables)
+  will allow users to override these values.
+ */
+.ui.vocabulary_cf {
+  h2.ui.header {
+    font-size: 1.2em;
```
Comment (mirekys) [inline oarepo_vocabularies/ui/theme/assets/semantic-ui/less/oarepo_vocabularies_ui/definitions/vocabulary_cf.less:27]: How to do this better?

Comment (Ducica) [reply oarepo_vocabularies/ui/theme/assets/semantic-ui/less/oarepo_vocabularies_ui/definitions/vocabulary_cf.less]: @mirekys Same here, please have a look. It also includes refactoring of the form and serach app according to the new principle. For detail page search app, this was not possible, as we use direct search kit there, due to the fact that we don't want for search app to influence the URL of the page. Thanks

File: oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/form/app.js [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
-export const overriddenComponents = {
-    "FormApp.layout": VocabularyForm,
+export const componentOverrides = {
+  [`${overridableIdPrefix}.FormApp.layout`]: FormAppLayout,
+  [`${overridableIdPrefix}.FormFields.container`]: VocabularyFormFields,
+  [`${overridableIdPrefix}.FormActions.container`]: VocabularyFormControlPanel,
 };
 
-createFormAppInit(overriddenComponents);
+createFormAppInit({ componentOverrides: componentOverrides });
```
Comment (mirekys) [inline oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/form/app.js:13]: just { componentOverrides }

File: oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/form/components/VocabularyFormFields/VocabularyFormFields.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    formConfig: { vocabularyProps },
+  } = useFormConfig();
+  const { formConfig } = useFormConfig();
+  const editMode = _has(formConfig, "updateUrl");
+  const hasPropFields = _has(vocabularyProps, "props");
+  return (
+    <React.Fragment>
+      <Grid.Row className="rel-mb-2">
+        <Grid.Column width={16}>
+          <CurrentLocationInformation
```
Comment (mirekys) [inline oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/form/components/VocabularyFormFields/VocabularyFormFields.jsx:28]: Shouldn't be part for FormAppLayout? No form input field here :)

File: oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/search/app.js [hunk 4]
```diff
@@ -1,61 +1,29 @@
...
+  [`${overridableIdPrefix}.SearchApp.layout`]: SearchAppLayoutWithConfig,
 };
-const overriddenComponents = overrideStore.getAll();
-createSearchAppInit(
-    { ...defaultComponents, ...overriddenComponents },
-    true,
-    "invenio-search-config",
-    true
-);
+createSearchAppsInit({ componentOverrides: componentOverrides });
```
Comment (mirekys) [inline oarepo_vocabularies/ui/theme/assets/semantic-ui/js/oarepo_vocabularies_ui/search/app.js:29]: just { componentOverrides }

**Review by mirekys — CHANGES_REQUESTED** [review]

**Review by mirekys — APPROVED** [review]

**Pull request merged**